src/app/validators/custom-validators.ts

```diff
@@ -168,25 +168,33 @@ export class CustomValidators {
 
   // Set this on both password and confirmation fields so it runs when either changes
   // confirm should be true for the confirmation field validator
-  static matchPassword(match: string, confirm: boolean): ValidatorFn {
+  // match is true by default, for unmatching passwords, match should be false
+  static matchPassword(matchField: string, confirm: boolean, match: boolean = true): ValidatorFn {
 
     return (ac: AbstractControl) => {
       if (!ac.parent) {
         return null;
       }
 
-      const matchControl = ac.parent.get(match),
+      const matchControl = ac.parent.get(matchField),
         val1 = ac.value,
         val2 = matchControl.value,
         confirmControl: AbstractControl = confirm ? ac : matchControl;
 
       // If passwords do not match, set error for confirmation field
-      if (val1 !== val2) {
+      if (match && val1 !== val2) {
```

Comment:
Rather than adding another conditional, you only need to check if `match === (val1 !== val2)`.  You would then set an `errorType` variable with `matchPassword` or `unmatchPassword` and use that like so: `setErrors({ [errorType]: false })`.